core/src/main/java/org/elasticsearch/cluster/routing/allocation/command/AllocateEmptyPrimaryAllocationCommand.java

```diff
@@ -128,7 +130,8 @@ public RerouteExplanation execute(RoutingAllocation allocation, boolean explain)
         }
 
         initializeUnassignedShard(allocation, routingNodes, routingNode, shardRouting, unassignedInfoToUpdate, StoreRecoverySource.EMPTY_STORE_INSTANCE);
-
+        Logger logger = Loggers.getLogger(AllocateEmptyPrimaryAllocationCommand.class);
+        logger.warn("Allocating an empty primary for [" + index + "][" + shardId + "], legacy data might have been erased.");
```

Comment:
"legacy data might have been erased" is a bit weak. It is more like "this has erased all data in that shard." or something like that. Presumably the user has only executed this action if the shard is already unrecoverable.